From 04c6b8cb42cde5e1f70bd05445016f546ebc5a93 Mon Sep 17 00:00:00 2001
From: Nell <xianbin.zhu@spacemit.com>
Date: Tue, 3 Sep 2024 16:32:00 +0800
Subject: [PATCH 0978/1204] k1:thermal: fix the issue where the frequency
 cannot be increased to 1.8G after the temperature drops
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

1. for m1, the thermal governor will start-up when the temperature up to 85°C
	when the temperature drops to 80°C, 1.8G will be restored
2. for k1, the thermal governor will start-up when the temperature up to
   95°C like before

Change-Id: Ie4ad14e01ab38e662eb9f13d37c15d36aafacea1
---
 .../dts/spacemit/k1-x_thermal_cooling.dtsi    | 48 ++++++++++++++-----
 1 file changed, 36 insertions(+), 12 deletions(-)

diff --git a/arch/riscv/boot/dts/spacemit/k1-x_thermal_cooling.dtsi b/arch/riscv/boot/dts/spacemit/k1-x_thermal_cooling.dtsi
index e2d73dd2f6c3..b11d653af813 100644
--- a/arch/riscv/boot/dts/spacemit/k1-x_thermal_cooling.dtsi
+++ b/arch/riscv/boot/dts/spacemit/k1-x_thermal_cooling.dtsi
@@ -9,24 +9,30 @@ cluster0_thermal {
 
 		thermal0_trips: trips {
 			cls0_trip0: cls0-trip-point0 {
-				temperature = <90000>;
-				hysteresis = <4000>;
+				temperature = <75000>;
+				hysteresis = <5000>;
 				type = "passive";
 			};
 
 			cls0_trip1: cls0-trip-point1 {
-				temperature = <95000>;
+				temperature = <85000>;
 				hysteresis = <5000>;
 				type = "passive";
 			};
 
 			cls0_trip2: cls0-trip-point2 {
-				temperature = <105000>;
+				temperature = <95000>;
 				hysteresis = <5000>;
 				type = "passive";
 			};
 
 			cls0_trip3: cls0-trip-point3 {
+				temperature = <105000>;
+				hysteresis = <5000>;
+				type = "passive";
+			};
+
+			cls0_trip4: cls0-trip-point4 {
 				temperature = <115000>;
 				hysteresis = <5000>;
 				type = "critical";
@@ -36,6 +42,18 @@ cls0_trip3: cls0-trip-point3 {
 		cooling-maps {
 			map0 {
 				trip = <&cls0_trip0>;
+				cooling-device = <&cpu_0 0 0>,
+						 <&cpu_1 0 0>,
+						 <&cpu_2 0 0>,
+						 <&cpu_3 0 0>,
+						 <&cpu_4 0 0>,
+						 <&cpu_5 0 0>,
+						 <&cpu_6 0 0>,
+						 <&cpu_7 0 0>;
+			};
+
+			map1 {
+				trip = <&cls0_trip1>;
 				cooling-device = <&cpu_0 1 1>,
 						 <&cpu_1 1 1>,
 						 <&cpu_2 1 1>,
@@ -46,8 +64,8 @@ map0 {
 						 <&cpu_7 1 1>;
 			};
 
-			map1 {
-				trip = <&cls0_trip1>;
+			map2 {
+				trip = <&cls0_trip2>;
 				cooling-device = <&cpu_0 2 3>,
 						 <&cpu_1 2 3>,
 						 <&cpu_2 2 3>,
@@ -58,8 +76,8 @@ map1 {
 						 <&cpu_7 2 3>;
 			};
 
-			map2 {
-				trip = <&cls0_trip2>;
+			map3 {
+				trip = <&cls0_trip3>;
 				cooling-device = <&cpu_0 4 5>,
 						 <&cpu_1 4 5>,
 						 <&cpu_2 4 5>,
@@ -79,24 +97,30 @@ cluster1_thermal {
 
 		thermal1_trips: trips {
 			cls1_trip0: cls1-trip-point0 {
-				temperature = <90000>;
-				hysteresis = <4000>;
+				temperature = <75000>;
+				hysteresis = <5000>;
 				type = "passive";
 			};
 
 			cls1_trip1: cls1-trip-point1 {
-				temperature = <95000>;
+				temperature = <85000>;
 				hysteresis = <5000>;
 				type = "passive";
 			};
 
 			cls1_trip2: cls1-trip-point2 {
-				temperature = <105000>;
+				temperature = <95000>;
 				hysteresis = <5000>;
 				type = "passive";
 			};
 
 			cls1_trip3: cls1-trip-point3 {
+				temperature = <105000>;
+				hysteresis = <5000>;
+				type = "passive";
+			};
+
+			cls1_trip4: cls1-trip-point4 {
 				temperature = <115000>;
 				hysteresis = <5000>;
 				type = "critical";
-- 
2.47.0

